[FLINK-27879][flink-web] Translate the Documentation Style Guide page into Chinese (#548)
apache/flink-web: contributing/docs-style.zh.md

---
title:  "文档样式指南"
---

本指南概述了在编辑以及贡献 Flink 文档中必要的样式原则。目的是在你的贡献之旅中可以投入更好的社区精力去改进和扩展既有文档，并使其更 **易读**、 **一致** 和 **全面**。

{% toc %}

## 语言

> Flink 同时维护了 **英语** 和 **中文** 两种文档，当你拓展或者更新文档时，需要在 pull request 中包含两种语言版本。如果你不熟悉中文，确保本次贡献补充了如下额外操作：

Reviewer: ```suggestion
Flink 同时维护了 **英文** 和 **中文** 两种文档，当你拓展或者更新文档时，需要在 pull request 中包含两种语言版本。如果你不熟悉中文，确保本次贡献补充了如下额外操作：
```

Author: 已修正

Author: 已修正，待后续问题确认完一起提交。

> * 开一个翻译的 [JIRA](https://issues.apache.org/jira/projects/FLINK/issues/FLINK-12070?filter=allopenissues) 票据，并打上 chinese-translation 的标签；

Reviewer: `ticket` 在一些翻译中被翻译成`请求单`，如果比较难翻译的话，可以保持原始的`ticket`描述

Author: `ticket` 并不属于 JIRA 中特有的专业术语，能合理翻译应该还是翻译掉好一点，的确 `请求单` 更贴切一点

Reviewer: 嗯，如果这样子的话，还是需要翻译一下，我看目前还是保持着 `票据`的翻译。

* 在此票据上添加到原始 JIRA 票据的链接。

正在寻求有助于将现有文档翻译成中文的风格指南？请继续查阅 [这个翻译规范](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications)。

[回到顶部](#top)

## 语言风格

如下，你可以看到一些初步的原则，这些原则可以确保书写中的可读性和通俗易懂。如果想更深入、更细致的了解语言风格，也可以参考 [通用准则](#general-guiding-principles)。

### 语态和语气

> * **使用主动语态。**[主动语态](https://medium.com/@DaphneWatson/technical-writing-active-vs-passive-voice-485dfaa4e498)简洁，并让内容更具有吸引力。如果你在句子的动词后添加 _by zombies_ 后仍然读的通，那么你用的就是被动语态。

Reviewer: `If you add _by zombies_
  after the verb in a sentence and it still makes sense, you are using the
  passive voice.` 这句是阐释了英文中的被动语态，无法直接套用在中文中，建议换成中文的例子。

Author: - 主体是英文还是中文
这里当时也纠结过。
当时的想法是主体应该是英文：即这个文档规范主要约束的应该是写英文文档（而相对于汉藏语系，被动语态的确也是英语语系用的多），因为中文应该都是翻译过来的。
而翻译的语态问题在前面的一个翻译建议的 cwiki 链接中明确了，所以这里就没怎么动了。

- 按照中文描述
而中文的被动语态，试了几个，文中的鉴别方法应该也是适用的，不过要根据中文调整为：在动词前加“被”后也能语句通畅。
同时英文文档中例子也可以直接用，所以也可以这样：
```
* **使用主动语态。**[主动语态](https://medium.com/@DaphneWatson/technical-writing-active-vs-passive-voice-485dfaa4e498)简洁，并让内容更具有吸引力。如果你在句子的动词前添加 _被僵尸_  后仍然读的通，那么就是被动语态。

  <div class="alert alert-info">
    <b>主动语态</b>
    <p>"你可以在 IDE 或者命令行中运行这个例子。"</p>

    <p></p>

    <b>被动语态</b> 
    <p>"这个例子可以在 IDE 或者命令行中（被僵尸）运行。"</p>
  </div>
```

Reviewer: 我想了想，这个文档的目的还是以写出好的英文文档。中文文档不是Flink社区的第一语言，所以面向的对象确实是如何写出优秀的英文文档。关于 `by the zombines`的说法，应该是英文俚语，我看很多翻译书籍中也有类似的表达：[清晰簡明的英文寫作指南](https://books.google.com.hk/books?id=WkEyEAAAQBAJ&pg=PA8&lpg=PA8&dq=by+zombies+%E8%A2%AB%E5%8A%A8%E8%AF%AD%E6%80%81&source=bl&ots=sJb76uNdna&sig=ACfU3U2T6wB2A1lDS7yR_qwMylItfvw-Hw&hl=zh-CN&sa=X&redir_esc=y&sourceid=cndr#v=onepage&q=by%20zombies%20%E8%A2%AB%E5%8A%A8%E8%AF%AD%E6%80%81&f=false)

  <div class="alert alert-info">
    <b>主动语态</b>
    <p>"You can run this example in your IDE or on the command line."</p>

    <p></p>

    <b>被动语态</b> 
    <p>"This example can be run in your IDE or on the command line (by zombies)."</p>
  </div>

* **使用你，而不是我们。** 用 _我们_ 会让用户感到困惑以及傲慢，给人“我们是一个秘密组织的成员，而 _你_ 并没有获得会员邀请”的感觉。所以用 _你_ 来建议用户。

* **避免使用针对性别和文化的语言。**文档无需指定性别：技术写作应当 [性别中立](https://techwhirl.com/gender-neutral-technical-writing/)。还有，在你的文化和日常交流中被认为是理所应当的行话和惯例，在其他地方可能很不一样。幽默就是很典型的例子：在某个文化中很棒的笑话，但在其他文化中可能被广泛误解。

* **避免对操作做能力限定以及对难度提前下结论。**对于很艰难才能完成操作或者操作中很容易沮丧的用户，使用诸如 _快速_ 或者 _容易_ 是糟糕的文档体验。

* **避免使用大写单词**来突出或者强调陈述。使用例如 **加粗** 或者 _斜体_ 来突出关键词通常会更礼貌。如果一个不明显的声明需要突出以引起更多的注意，可以按照段落分组，段落以标签开头，配合对应的 HTML 标记来突出显示：
  * `<span class="label label-info">Note</span>`
  * `<span class="label label-warning">Warning</span>`
  * `<span class="label label-danger">Danger</span>`

### 使用 Flink 特定术语

使用清晰的术语定义，也可以对要表达的内容提供有帮助的资源链接来辅助说明，例如其他的文档页面或者 [Flink 术语表]({{site.DOCS_BASE_URL}}flink-docs-stable/docs/concepts/glossary)。目前，术语表仍在编辑中，新术语可以开 pull-request 来提交。

[回到顶部](#top)

> ## 仓库

Reviewer: ```suggestion
## 代码仓库
```

Author: 已明确，待前面问题确认完一起提交。

    <th></th>
    <th style="vertical-align : middle;"><center><b>变量名</b></center></th>
    <th style="vertical-align : middle;"><center><b>可能值</b></center></th>
    <th style="vertical-align : middle;"><center><b>描述</b></center></th>
  </tr>
  <tr>
    <td><b>布局</b></td>
    <td>layout</td>
    <td>{base,plain,redirect}</td>
    <td>要使用的布局文件。布局文件位于 <i>_layouts</i> 目录下。</td>
  </tr>
  <tr>
    <td><b>内容</b></td>
    <td>title</td>
    <td>%s</td>
    <td>此标题是页面最顶部 （1级） 的标题。</td>
  </tr>
  <tr>
	    <td rowspan="4" style="vertical-align : middle;"><b>导航</b></td>
	    <td>nav-id</td>
	    <td>%s</td>
	    <td>页面 ID。其他页面可以使用此 ID 作为他们的 nav-parent_id。</td>
	  </tr>
	  <tr>
	    <td>nav-parent_id</td>
	    <td>{root,%s}</td>
	    <td>页面父级 ID。最低导航级别为 root。</td>
	  </tr>
	  <tr>
	    <td>nav-pos</td>
	    <td>%d</td>
	    <td>在每个导航级别下页面的相对位置。</td>
	  </tr>
	  <tr>
	    <td>nav-title</td>
	    <td>%s</td>
	    <td>此标题用于重载默认的文本链接（标题）</td>
	  </tr>
 </thead>
</table>
</font>

文档范围的信息和配置位于 `_config.yml` 下，在前言中也是可用的，通过 site 变量使用。可以使用以下语法访问这些设置：

```liquid
{{ "{{ site.CONFIG_KEY " }}}}
```
当生成文档时，占位符会被替换成变量 `CONFIG_KEY` 的值。

[回到顶部](#top)

## 格式化

以下各节列出了基本的格式准则，可以帮助你开始编写一致且易于浏览的文档。

### 标题

> 在 Markdown 中，标题是任意以井号（#）开头的行，井号的数量表示标题级别。标题永远是嵌套和连续的--永远不能因为样式原因跳过标题级别！

Reviewer: ```suggestion
在 Markdown 中，标题是任意以井号（#）开头的行，井号的数量表示标题级别。标题永远是嵌套和连续的，不能因为样式原因跳过标题级别！
```

Author: 再读一遍后，结合前面已经明确主语是标题的情况下，直接改为 `，不能因为样式原因跳级` ，是不是更通顺点？

Reviewer: `跳级`在一般语义下，有“升学跳级”的含义，感觉还是要避免歧义，多一些描述。

  ```

* 发出陷阱危险信号，或提醒关注必须要遵循的重要信息：

  ```html
  <div class="alert alert-danger"> // Danger Message </div>
  ```

### 链接

添加文档链接是一种有效的方法--可以引导用户更好的理解正在讨论的主题，而不会有重写的风险。

* **指向页面中各节的链接。** 每个标题都会生成一个隐式标识符，以便在页面中直接跳转。此标识符是通过将标题设置为小写并将内部空格替换为连字符来生成的。

  * **标题：** ## Heading Title
  * **ID：** #heading-title
  <p></p>

  ```liquid 
  [Link Text](#heading-title) 
  ```

* **链接到 Flink 文档的其他页面。**

  {% raw %}
  ```liquid 
  [Link Text]({% link path/to/link-page.md %})
  ```
  {% endraw %}

* **指向外部页面的链接**

  ```liquid 
  [Link Text](external_url)
  ```

<div class="alert alert-warning">
  <h3>最佳实践</h3>

  链使用可以提供操作或者目标信息的描述性链接名称。例如，避免使用“了解更多”或“单击此处”链接。
</div>

### 可视化内容

图形和其他可视化内容放置在根目录的 _fig_ 目录下，可以使用类似于链接的语法在文档页面中引用：

```liquid 
![Picture Text]({{ "{{ site.baseurl " }}}}/fig/image_name.png){:height="200px" width="200px"}
```

或者使用纯 HTML：

{% highlight html %}
<img src="{{ site.baseurl }}/fig/image_name.png" class="center" height="200px" width="200px"/>
{% endhighlight %}

<div class="alert alert-warning">
  <h3>最佳实践</h3>

  在适当或必要的情况下使用流程图、表格和图形进行额外说明，但切勿作为独立的信息来源。确保内容中的字体大小不影响阅读，并且整体分辨率足够。
</div>

### 代码

* **行内代码、** 使用包围的反引号（ **\`**）来高亮正常文本流中的小代码段或者语言结构类型的引用。

> * **代码块。** 表示自包含示例、功能演练、最佳实践演示或其他有用场景的代码，应使用带有适当 [语法突出]((https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers))显示的围栏代码块（fenced code block）进行包装。实现此目的的一种方式是标记：

Reviewer: ```suggestion
* **代码块。** 表示自包含示例、功能演练、最佳实践演示或其他有用场景的代码，应使用带有适当 [语法高亮]((https://github.com/rouge-ruby/rouge/wiki/List-of-supported-languages-and-lexers))显示的围栏代码块（fenced code block）进行包装。其中一种代码块实现方式如下：
```

Author: 已修改，待前面问题确认完一起提交。
